src/lib/logstore/log_store_service.cpp

```diff
@@ -260,7 +260,7 @@ void LogStoreService::start_threads() {
 
     m_flush_fiber = nullptr;
     iomanager.create_reactor("log_flush_thread", iomgr::TIGHT_LOOP | iomgr::ADAPTIVE_LOOP, 1 /* num_fibers */,
-                             [this, &ctx](bool is_started) {
+                             [this, ctx](bool is_started) {
```

Comment:
This is needed though there is sync wait in this same function on this ctx. Because this lambda is being stored in iomgr and during IORactor destroy, this function is also being triggered to notify this thread is being destroyed, e.g. is_started set to false.

Reply:
Ack fixed in L276